include/Dialect/LWE/IR/LWEAttributes.td

```diff
@@ -55,10 +67,165 @@ def LWE_BitFieldEncoding: LWE_EncodingAttr<"BitFieldEncoding", "bit_field_encodi
+    powers of the root of unity, $\omega, \omega^3, \dots, \omega^{2N-1}$. For
+    polynomials in the slightly larger space $\mathbb{R}[x] / (x^N + 1)$, the
+    image of the canonical embedding is the subspace $H \subset \mathbb{C}^N$
+    defined by tuples $(z_1, \dots, z_N)$ such that $\overline{z_i} =
+    \overline{z_{N-i+1}}$. Note that this property holds because polynomial
+    evaluation commutes with complex conjugates, and the second half of the
+    roots of unity evaluate are complex conjugates of the first half. The
+    converse, that any such tuple with complex conjugate symmetry has an
+    inverse under $\tau$ with all real coefficients, makes $\tau$ is a
+    bijection onto $H$. $\tau$ and its inverse are explicitly computable as
+    discrete Fourier Transforms.
+
+    Because of the symmetry in canonical embedding for real polynomials, inputs
+    to this encoding can be represented as a list of $N/2$ complex points, with
+    the extra symmetric structure left implicit. $\tau_N$ and its inverse can
+    also be explicitly computed without need to expand the vectors to length
+    $N$.
+
+    The rounding step is required to invert the decoding because, while
+    cleartexts must be (implicitly) in the subspace $H$, they need not be the
+    output of $\tau_N$ for an _integer_ polynomial. The rounding step ensures
+    we can use integer polynomial plaintexts for the FHE operations. There are
+    multiple rounding mechanisms, and this attribute does not specify which is
+    used, because in theory two ciphertexts that have used different roundings
+    are still compatible, though they may have different noise growth patterns.
+
+    The scaling parameter $\Delta$ is specified by the `cleartext_start` and
+    `cleartext_bitwidth` parameters, which are applied coefficient-wise using
+    the same semantics as the `bit_field_encoding`.
+
+    This attribute can be used in multiple ways:
+
+    - On a `poly.poly`, it asserts that the polynomial has been transformed
+      from a coefficient list using the canonical embedding.
+    - On a tensor of `poly.poly`, it asserts that the tensor is an RLWE
+      ciphertext for some RLWE scheme that supports the approximate embedding
+      encoding.
+
+    A typical flow for the CKKS scheme using this encoding would be to apply an
+    inverse FFT operation to invert the canonical embedding to be a polynomial
+    with real coefficients, then encrypt scale the resulting polynomial's
+    coefficients according to the scaling parameters, then round to get integer
+    coefficients.
+
+    Example:
+
+    ```
+    #generator = #poly.polynomial<1 + x**1024>
+    #ring = #poly.ring<cmod=65536, ideal=#generator>
+    #lwe_encoding = #lwe.poly_evaluation_encoding<cleartext_start=30, cleartext_bitwidth=3>
+
+    %evals = arith.constant <[1, 2, 4, 5]> : tensor<4xi16>
+    // Note no `intt` operation exists in poly yet.
```

Comment:
You may want to label this with a TODO(issue) just so we can make sure to fixup docs when it is implemented.

Reply:
Done